examples/api/lib/material/segmented_button/segmented_button.0.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for [SegmentedButton].
+
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(const SegmentedButtonApp());
+}
+
+class SegmentedButtonApp extends StatelessWidget {
+  const SegmentedButtonApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(useMaterial3: true),
+      home: Scaffold(
+        body: Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: const <Widget>[
+              Spacer(),
+              Text('Single choice'),
+              SingleChoice(),
+              SizedBox(height: 20),
+              Text('Multiple choice'),
+              MultipleChoice(),
+              Spacer(),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+enum Calendar { day, week, month, year }
+
+class SingleChoice extends StatefulWidget {
+  const SingleChoice({super.key});
+
+  @override
+  State<SingleChoice> createState() => _SingleChoiceState();
+}
+
+class _SingleChoiceState extends State<SingleChoice> {
+
+  Calendar _selected = Calendar.day;
+
+  @override
+  Widget build(BuildContext context) {
+    return SegmentedButton<Calendar>(
+      segments: const <ButtonSegment<Calendar>>[
+        ButtonSegment<Calendar>(value: Calendar.day, label: Text('Day'), icon: Icon(Icons.calendar_view_day)),
+        ButtonSegment<Calendar>(value: Calendar.week, label: Text('Week'), icon: Icon(Icons.calendar_view_week)),
+        ButtonSegment<Calendar>(value: Calendar.month, label: Text('Month'), icon: Icon(Icons.calendar_view_month)),
+        ButtonSegment<Calendar>(value: Calendar.year, label: Text('Year'), icon: Icon(Icons.calendar_today)),
+      ],
+      selected: <Calendar>{_selected},
+      onSelectionChanged: (Set<Calendar> selected) {
+        setState(() {
+          _selected = selected.first;
```

Comment:
It might be simpler to just carry the Set through the implementation, as in the MultipleChoice example. Also, some  naming micro suggestions.
```dart
// By default the SegmentedButton only allows one item to be
// selected at a time. The value of that item is selection.first.
Calendar selection = {Calendar.day};

 onSelectionChanged: (Set<Calendar> newSelection) {
   setState(() {
     selection = newSelection;
   }
 }
```

Reply:
Agreed on the naming issue, but I think it makes sense to show a single value state for the single choice example as that is probably a more common use case and storing the value as a set of one value seems confusing.

Reply:
OK, thanks for considering it.

Reply:
No worries. If you feel strongly about it, I can definitely switch it. I just thought it would be good to show how to deal with the two main uses in ways most apps probably need.